src/main/java/org/jabref/logic/importer/fileformat/MarcXmlParser.java

```diff
@@ -437,6 +447,21 @@ private void putIssue(BibEntry bibEntry, Element datafield) {
         }
     }
 
+    private void putThesisDetails(BibEntry bibEntry, Element datafield) {
+        String thesisDegree = getSubfield("b", datafield);
+        String school = getSubfield("c", datafield);
+
+
+        if ("Masterarbeit".equals(thesisDegree) & StringUtil.isNotBlank(school)) {
```

Comment:
Always use && for Boolean.

I think, the condition should be separated. The master Thesis type should be set in all cases

Reply:
Ok, thank you. 

Even if it is a bachelor's thesis, a habilitation or a dissertation?
I do not need to use the Standard types thesis or PhDThesis at all?

Reply:
> Ok, thank you. Even if it is a bachelor's thesis, a habilitation or a dissertation?
> 
> Is thesis a general term or is it exclsuive for bachelor's thesis? I guess it makes sense as you rarely need to cite a bachelor's thesis. I update it in my next commit.

Under the tag 502 you will find the type of thesis. A distinction is made here between a dissertation, a master's thesis, a bachelor's thesis and a habilitation thesis. The standard types PhD thesis, master's thesis and thesis are already implemented. Should I then generally mark a thesis entry as master's thesis and a bachelor's thesis a thesis, and a dissertation as a PhD thesis?